src/main/java/io/getunleash/repository/FeatureRepository.java

```diff
@@ -0,0 +1,100 @@
+package io.getunleash.repository;
+
+import io.getunleash.FeatureToggle;
+import io.getunleash.Segment;
+import io.getunleash.UnleashException;
+import io.getunleash.event.EventDispatcher;
+import io.getunleash.event.UnleashReady;
+import io.getunleash.lang.Nullable;
+import io.getunleash.util.UnleashConfig;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FeatureRepository implements IFeatureRepository {
+    private final FeatureBackupHandlerFile featureBackupHandler;
+    private final FeatureBootstrapHandler featureBootstrapHandler;
+    private final HttpFeatureFetcher featureFetcher;
+    private final EventDispatcher eventDispatcher;
+
+    private FeatureCollection featureCollection;
+    private boolean ready;
+
+    private static FeatureRepository instance = null;
+
+    private FeatureRepository(UnleashConfig unleashConfig) {
+        this.featureBackupHandler = FeatureBackupHandlerFile.getInstance();
+        this.featureFetcher = HttpFeatureFetcher.getInstance();
+        this.featureBootstrapHandler = FeatureBootstrapHandler.getInstance();
+
+        this.eventDispatcher = new EventDispatcher(unleashConfig);
+        this.featureCollection = this.featureBackupHandler.read();
+        if (this.featureCollection.getToggleCollection().getFeatures().isEmpty()) {
+            this.featureCollection = this.featureBootstrapHandler.read();
+        }
+
+        if (unleashConfig.isSynchronousFetchOnInitialisation()) {
+            updateFeatures().run();
+        }
+
+        unleashConfig
+                .getScheduledExecutor()
+                .setInterval(updateFeatures(), 0, unleashConfig.getFetchTogglesInterval());
+    }
+
+    public static synchronized FeatureRepository getInstance() {
+        if (instance == null) {
+            throw new AssertionError("FeatureRepository:: You have to call init first");
```

Comment:
Since the API already has a good fallback (evaluating features as false) when the Unleash API is not reachable, then I can see avoiding a 'fast fail' could be acceptable. 

I like fast failure so I can fix it quicker though. And not be scratching my head wondering why the toggles aren't working (and I've overlooked a single log message)

Reply:
Mixed feelings on this one but I think this is okay. It's pretty easy to setup some of the SDKs wrong and then never get toggles resolving correctly, it's super frustrating when that happens and small comfort when I realise I'm being dumb. I think so long as the error can only happen during the setup code it's okay, it'd be a huge worry for me if the SDK could crash its host process later. I guess this is the choice between fail-fast or never-work-at-all